provider/test_utils.go

```diff
@@ -0,0 +1,197 @@
+	authType          string
+	env               map[string]string
+	assertError       string
+	assertAuth        string
+	assertHost        string
+	assertAzure       bool
+}
+
+func (tt providerFixture) rawConfig() map[string]string {
+	rawConfig := map[string]string{}
+	if tt.host != "" {
+		rawConfig["host"] = tt.host
+	}
+	if tt.token != "" {
+		rawConfig["token"] = tt.token
+	}
+	if tt.username != "" {
+		rawConfig["username"] = tt.username
+	}
+	if tt.password != "" {
+		rawConfig["password"] = tt.password
+	}
+	if tt.configFile != "" {
+		rawConfig["config_file"] = tt.configFile
+	}
+	if tt.profile != "" {
+		rawConfig["profile"] = tt.profile
+	}
+	if tt.azureClientID != "" {
+		rawConfig["azure_client_id"] = tt.azureClientID
+	}
+	if tt.azureClientSecret != "" {
+		rawConfig["azure_client_secret"] = tt.azureClientSecret
+	}
+	if tt.azureTenantID != "" {
+		rawConfig["azure_tenant_id"] = tt.azureTenantID
+	}
+	if tt.azureResourceID != "" {
+		rawConfig["azure_workspace_resource_id"] = tt.azureResourceID
+	}
+	if tt.authType != "" {
+		rawConfig["auth_type"] = tt.authType
+	}
+	return rawConfig
+}
+
+func (tt providerFixture) rawConfigSDKv2() map[string]any {
+	rawConfig := tt.rawConfig()
+	rawConfigSDKv2 := map[string]any{}
+	for k, v := range rawConfig {
+		rawConfigSDKv2[k] = v
+	}
+	return rawConfigSDKv2
+}
+
+func (tt providerFixture) rawConfigPluginFramework() tftypes.Value {
+	rawConfig := tt.rawConfig()
+
+	rawConfigTypeMap := map[string]tftypes.Type{}
+	for k := range rawConfig {
+		rawConfigTypeMap[k] = tftypes.String
+	}
+	rawConfigType := tftypes.Object{
+		AttributeTypes: rawConfigTypeMap,
+	}
+
+	rawConfigValueMap := map[string]tftypes.Value{}
+	for k, v := range rawConfig {
+		rawConfigValueMap[k] = tftypes.NewValue(tftypes.String, v)
+	}
+	rawConfigValue := tftypes.NewValue(rawConfigType, rawConfigValueMap)
+	return rawConfigValue
+}
+
+func (tc providerFixture) applyWithSDKv2(t *testing.T) *common.DatabricksClient {
+	c, err := configureProviderAndReturnClient_SDKv2(t, tc)
+	return tc.applyAssertions(c, t, err)
+}
+
+func (tc providerFixture) applyWithPluginFramework(t *testing.T) *common.DatabricksClient {
+	c, err := configureProviderAndReturnClient_PluginFramework(t, tc)
+	return tc.applyAssertions(c, t, err)
+}
+
+func (tc providerFixture) applyAssertions(c *common.DatabricksClient, t *testing.T, err error) *common.DatabricksClient {
+	if tc.assertError != "" {
+		require.NotNilf(t, err, "Expected to have %s error", tc.assertError)
+		require.True(t, strings.HasPrefix(err.Error(), tc.assertError),
+			"Expected to have '%s' error, but got '%s'", tc.assertError, err)
+		return nil
+	}
+	if err != nil {
+		require.NoError(t, err)
+		return nil
+	}
+	assert.Equal(t, tc.assertAzure, c.IsAzure())
+	assert.Equal(t, tc.assertAuth, c.Config.AuthType)
+	assert.Equal(t, tc.assertHost, c.Config.Host)
+	return c
+}
+
+func (tc providerFixture) apply(t *testing.T) {
+	_ = tc.applyWithSDKv2(t)
+	_ = tc.applyWithPluginFramework(t)
```

Comment:
Can we do:
```go
t.Run("sdkv2", func() { _ = tc.applyWithSDKv2(t) })
t.Run("plugin framework", func() { _ = tc.applyWithPluginFramework(t) })
```
This treats these two cases as different tests, so we can see more precisely which one fails.